packages/webapp/pages/account/notifications.tsx

```diff
@@ -0,0 +1,194 @@
+        data,
+      }),
+    { onSuccess: refetchGeneralData },
+  );
+
+  const { data: deviceData, refetch: refetchDeviceData } = useQuery(
+    ['device_preferences', visit.sessionId],
+    () =>
+      request<DevicePreferenceData>(
+        `${apiUrl}/graphql`,
+        DEVICE_PREFERENCE_QUERY,
+        { deviceId: visit.sessionId },
+      ),
+  );
+  const { mutateAsync: updateDevicePreference } = useMutation(
+    (data: Omit<DeviceNotificationPreference, 'deviceId' | 'description'>) =>
+      request(`${apiUrl}/graphql`, UPDATE_DEVICE_PREFERENCE_MUTATION, {
+        data,
+        deviceId: visit.sessionId,
+      }),
+    { onSuccess: refetchDeviceData },
+  );
+
+  const { preference: generalPreference } = generalData ?? {};
+  const { marketingEmail, notificationEmail } = generalPreference ?? {};
+  const { preference: devicePreference } = deviceData ?? {};
+  const { pushNotification } = devicePreference ?? {};
+  const [emailNotification, setEmailNotification] = useState(
+    marketingEmail || notificationEmail,
+  );
+  const onToggleEmailNotification = () => {
+    const value = !emailNotification;
+    updateGeneralPreference({
+      marketingEmail: value,
+      notificationEmail: value,
+    });
+    setEmailNotification(value);
+  };
+
+  return (
+    <AccountPageContainer title="Notifications">
+      <div className="flex flex-row">
+        <AccountContentSection
+          className={{
+            heading: 'mt-0',
+            container: 'flex flex-col flex-1 w-full',
+          }}
+          title="Push notifications"
+          description="The daily.dev notification system notifies you of important events such as replies, mentions, updates, etc."
+        />
+        <div className="mx-4 w-px h-full bg-theme-divider-tertiary" />
+        <Switch
+          data-testId="push_notification-switch"
+          data-testValue={pushNotification}
+          inputId="push_notification-switch"
+          name="push_notification"
+          className="w-20"
+          compact={false}
+          checked={pushNotification}
+          onToggle={() =>
+            updateDevicePreference({ pushNotification: !pushNotification })
+          }
+        >
+          {pushNotification ? 'On' : 'Off'}
+        </Switch>
+      </div>
+      {isAlertShown && (
+        <div className="relative mt-6 w-full rounded-16 border border-theme-color-cabbage">
+          <Pointer
+            className="absolute -top-5 right-8"
+            color={PointerColor.Cabbage}
+          />
+          <div className="flex overflow-hidden relative flex-row p-4">
+            <p className="flex-1 break-words typo-subhead">
+              Switch on push notifications so you never miss exciting
+              discussions, upvotes, replies or mentions on daily.dev!
+            </p>
+            <img
+              className="absolute top-0 right-10"
+              src={cloudinary.notifications.browser}
+              alt="Preview"
```

Comment:
Perhaps we should inline this to the other alt we used in the previous pr?

Reply:
Yep. Right way to go. Let me refactor this.